api/utils/graphs.ts

```diff
@@ -0,0 +1,15 @@
+import { IActionProto } from "df/core/session";
+import { dataform } from "df/protos/ts";
+
+export function actionsByStringifiedTarget(compiledGraph: dataform.ICompiledGraph) {
```

Comment:
The StringifiedMap class is open sourced now, and will arguably be more correct as it factors in ordering of keys, in case we somehow mess that up in future.

return new StringifiedMap(JSONObjectStringifier.create(), ...entries);